Dockerfile

```diff
@@ -1,13 +1,23 @@
-FROM golang:1.20-alpine as builder
+# syntax=docker/dockerfile:1
+# Build
+FROM --platform=$BUILDPLATFORM golang:1.20-alpine AS builder
+
+ARG TARGETOS TARGETARCH
+
+ENV GOOS=$TARGETOS \
+  GOARCH=$TARGETARCH \
+  CGO_ENABLED=0
+
 WORKDIR $GOPATH/src/go.k6.io/k6
 COPY . .
 RUN apk --no-cache add git=~2
-RUN CGO_ENABLED=0 go install -a -trimpath -ldflags "-s -w -X go.k6.io/k6/lib/consts.VersionDetails=$(date -u +"%FT%T%z")/$(git describe --tags --always --long --dirty)"
+RUN go build -a -trimpath -ldflags "-s -w -X go.k6.io/k6/lib/consts.VersionDetails=$(date -u +"%FT%T%z")/$(git describe --tags --always --long --dirty)" -o /usr/bin/k6 .
```

Comment:
Is there a specific reason for writing the output binary to the `/usr/bin` folder instead of `/go/bin`?

Reply:
That `/usr/bin` will always exist, even if the golang container image maintainers decide to change the $GOPATH in the future. As I explained originally, I would prefer `/opt/bin` since is FHS-compatible but for a temporary stage, `/usr/bin` will be more than enough unless the k6 binary starts becoming standard-issue on distributions :)

Reply:
`/usr/local/bin` might be safer in that case, but since we already place it in `/usr/bin` in the final image, this is fine.

I would prefer changing the `WORKDIR` to somewhere outside of `$GOPATH/src`, which is a remnant of Go's dark ages before Go modules :face_exhaling:. Changing it to e.g. `/opt/k6` would ensure the directory exists, and we could write the binary there. But that's out of scope for this PR and can be done later.